src/main/ApplicationImpl.cpp

```diff
@@ -308,6 +308,9 @@ ApplicationImpl::initialize(bool createNewDB, bool forceRebuild)
         upgradeToCurrentSchemaAndMaybeRebuildLedger(true, forceRebuild);
     }
 
+    // Subtle: restore persistent state after a potential DB schema upgrade
+    mPersistentState->restoreState();
```

Comment:
Just to double-check: this should be safe for restarts without upgrades, right?

Reply:
Yeah, when a node upgrades its version to include this change, it'll do a schema upgrade, which will replace all of the old-style rows with new ones. If there is no upgrade, it means the schema is already up-to-date, which is what `restoreState` expects (as it queries PersistentStateV1)